tests/Unit/Rules/ValidTimezoneTest.php

```diff
@@ -33,3 +33,19 @@
     'Europe/Londonn',
     'Asia/Tokyo0',
 ])->fails();
+
+test('a deprecated timezone must be valid attempt 1', function () {
+    $response = $this->post(route('profile.timezone.update'), [
+        'timezone' => 'Asia/Calcutta',
+    ]);
+
+    $response->assertStatus(200);
+});
+
+test('a deprecated timezone must be valid attempt 2', function () {
+    $response = $this->post(route('profile.timezone.update'), [
+        'timezone' => 'Asia/Katmandu',
+    ]);
+
+    $response->assertStatus(200);
+});
```

Comment:
```suggestion
test('support for deprecated timezones', function () {
    $rule = new ValidTimezone();

    $fail = fn (string $errorMessage) => $this->fail($errorMessage);

    $rule->validate('timezone', $timezone, $fail);

    expect(true)->toBeTrue();
})->with([
'Asia/Calcutta',
'Asia/Katmandu',
'America/Godthab',
'US/Pacific-New',
]);

Reply:
`US/Pacific-New` might not pass I guess. I would check.

Reply:
Yes it is not. Instead this is the list:

```
array:600 [ // app/Rules/ValidTimezone.php:36
  0 => "Africa/Abidjan"
  1 => "Africa/Accra"
  2 => "Africa/Addis_Ababa"
  3 => "Africa/Algiers"
  4 => "Africa/Asmara"
  5 => "Africa/Asmera"
  6 => "Africa/Bamako"
  7 => "Africa/Bangui"
  8 => "Africa/Banjul"
  9 => "Africa/Bissau"
  10 => "Africa/Blantyre"
  11 => "Africa/Brazzaville"
  12 => "Africa/Bujumbura"
  13 => "Africa/Cairo"
  14 => "Africa/Casablanca"
  15 => "Africa/Ceuta"
  16 => "Africa/Conakry"
  17 => "Africa/Dakar"
  18 => "Africa/Dar_es_Salaam"
  19 => "Africa/Djibouti"
  20 => "Africa/Douala"
  21 => "Africa/El_Aaiun"
  22 => "Africa/Freetown"
  23 => "Africa/Gaborone"
  24 => "Africa/Harare"
  25 => "Africa/Johannesburg"
  26 => "Africa/Juba"
  27 => "Africa/Kampala"
  28 => "Africa/Khartoum"
  29 => "Africa/Kigali"
  30 => "Africa/Kinshasa"
  31 => "Africa/Lagos"
  32 => "Africa/Libreville"
  33 => "Africa/Lome"
  34 => "Africa/Luanda"
  35 => "Africa/Lubumbashi"
  36 => "Africa/Lusaka"
  37 => "Africa/Malabo"
  38 => "Africa/Maputo"
  39 => "Africa/Maseru"
  40 => "Africa/Mbabane"
  41 => "Africa/Mogadishu"
  42 => "Africa/Monrovia"
  43 => "Africa/Nairobi"
  44 => "Africa/Ndjamena"
  45 => "Africa/Niamey"
  46 => "Africa/Nouakchott"
  47 => "Africa/Ouagadougou"
  48 => "Africa/Porto-Novo"
  49 => "Africa/Sao_Tome"
  50 => "Africa/Timbuktu"
  51 => "Africa/Tripoli"
  52 => "Africa/Tunis"
  53 => "Africa/Windhoek"
  54 => "America/Adak"
  55 => "America/Anchorage"
  56 => "America/Anguilla"
  57 => "America/Antigua"
  58 => "America/Araguaina"
  59 => "America/Argentina/Buenos_Aires"
  60 => "America/Argentina/Catamarca"
  61 => "America/Argentina/ComodRivadavia"
  62 => "America/Argentina/Cordoba"
  63 => "America/Argentina/Jujuy"
  64 => "America/Argentina/La_Rioja"
  65 => "America/Argentina/Mendoza"
  66 => "America/Argentina/Rio_Gallegos"
  67 => "America/Argentina/Salta"
  68 => "America/Argentina/San_Juan"
  69 => "America/Argentina/San_Luis"
  70 => "America/Argentina/Tucuman"
  71 => "America/Argentina/Ushuaia"
  72 => "America/Aruba"
  73 => "America/Asuncion"
  74 => "America/Atikokan"
  75 => "America/Atka"
  76 => "America/Bahia"
  77 => "America/Bahia_Banderas"
  78 => "America/Barbados"
  79 => "America/Belem"
  80 => "America/Belize"
  81 => "America/Blanc-Sablon"
  82 => "America/Boa_Vista"
  83 => "America/Bogota"
  84 => "America/Boise"
  85 => "America/Buenos_Aires"
  86 => "America/Cambridge_Bay"
  87 => "America/Campo_Grande"
  88 => "America/Cancun"
  89 => "America/Caracas"
  90 => "America/Catamarca"
  91 => "America/Cayenne"
  92 => "America/Cayman"
  93 => "America/Chicago"
  94 => "America/Chihuahua"
  95 => "America/Ciudad_Juarez"
  96 => "America/Coral_Harbour"
  97 => "America/Cordoba"
  98 => "America/Costa_Rica"
  99 => "America/Creston"
  100 => "America/Cuiaba"
  101 => "America/Curacao"
  102 => "America/Danmarkshavn"
  103 => "America/Dawson"
  104 => "America/Dawson_Creek"
  105 => "America/Denver"
  106 => "America/Detroit"
  107 => "America/Dominica"
  108 => "America/Edmonton"
  109 => "America/Eirunepe"
  110 => "America/El_Salvador"
  111 => "America/Ensenada"
  112 => "America/Fort_Nelson"
  113 => "America/Fort_Wayne"
  114 => "America/Fortaleza"
  115 => "America/Glace_Bay"
  116 => "America/Godthab"
  117 => "America/Goose_Bay"
  118 => "America/Grand_Turk"
  119 => "America/Grenada"
  120 => "America/Guadeloupe"
  121 => "America/Guatemala"
  122 => "America/Guayaquil"
  123 => "America/Guyana"
  124 => "America/Halifax"
  125 => "America/Havana"
  126 => "America/Hermosillo"
  127 => "America/Indiana/Indianapolis"
  128 => "America/Indiana/Knox"
  129 => "America/Indiana/Marengo"
  130 => "America/Indiana/Petersburg"
  131 => "America/Indiana/Tell_City"
  132 => "America/Indiana/Vevay"
  133 => "America/Indiana/Vincennes"
  134 => "America/Indiana/Winamac"
  135 => "America/Indianapolis"
  136 => "America/Inuvik"
  137 => "America/Iqaluit"
  138 => "America/Jamaica"
  139 => "America/Jujuy"
  140 => "America/Juneau"
  141 => "America/Kentucky/Louisville"
  142 => "America/Kentucky/Monticello"
  143 => "America/Knox_IN"
  144 => "America/Kralendijk"
  145 => "America/La_Paz"
  146 => "America/Lima"
  147 => "America/Los_Angeles"
  148 => "America/Louisville"
  149 => "America/Lower_Princes"
  150 => "America/Maceio"
  151 => "America/Managua"
  152 => "America/Manaus"
  153 => "America/Marigot"
  154 => "America/Martinique"
  155 => "America/Matamoros"
  156 => "America/Mazatlan"
  157 => "America/Mendoza"
  158 => "America/Menominee"
  159 => "America/Merida"
  160 => "America/Metlakatla"
  161 => "America/Mexico_City"
  162 => "America/Miquelon"
  163 => "America/Moncton"
  164 => "America/Monterrey"
  165 => "America/Montevideo"
  166 => "America/Montreal"
  167 => "America/Montserrat"
  168 => "America/Nassau"
  169 => "America/New_York"
  170 => "America/Nipigon"
  171 => "America/Nome"
  172 => "America/Noronha"
  173 => "America/North_Dakota/Beulah"
  174 => "America/North_Dakota/Center"
  175 => "America/North_Dakota/New_Salem"
  176 => "America/Nuuk"
  177 => "America/Ojinaga"
  178 => "America/Panama"
  179 => "America/Pangnirtung"
  180 => "America/Paramaribo"
  181 => "America/Phoenix"
  182 => "America/Port-au-Prince"
  183 => "America/Port_of_Spain"
  184 => "America/Porto_Acre"
  185 => "America/Porto_Velho"
  186 => "America/Puerto_Rico"
  187 => "America/Punta_Arenas"
  188 => "America/Rainy_River"
  189 => "America/Rankin_Inlet"
  190 => "America/Recife"
  191 => "America/Regina"
  192 => "America/Resolute"
  193 => "America/Rio_Branco"
  194 => "America/Rosario"
  195 => "America/Santa_Isabel"
  196 => "America/Santarem"
  197 => "America/Santiago"
  198 => "America/Santo_Domingo"
  199 => "America/Sao_Paulo"
  200 => "America/Scoresbysund"
  201 => "America/Shiprock"
  202 => "America/Sitka"
  203 => "America/St_Barthelemy"
  204 => "America/St_Johns"
  205 => "America/St_Kitts"
  206 => "America/St_Lucia"
  207 => "America/St_Thomas"
  208 => "America/St_Vincent"
  209 => "America/Swift_Current"
  210 => "America/Tegucigalpa"
  211 => "America/Thule"
  212 => "America/Thunder_Bay"
  213 => "America/Tijuana"
  214 => "America/Toronto"
  215 => "America/Tortola"
  216 => "America/Vancouver"
  217 => "America/Virgin"
  218 => "America/Whitehorse"
  219 => "America/Winnipeg"
  220 => "America/Yakutat"
  221 => "America/Yellowknife"
  222 => "Antarctica/Casey"
  223 => "Antarctica/Davis"
  224 => "Antarctica/DumontDUrville"
  225 => "Antarctica/Macquarie"
  226 => "Antarctica/Mawson"
  227 => "Antarctica/McMurdo"
  228 => "Antarctica/Palmer"
  229 => "Antarctica/Rothera"
  230 => "Antarctica/South_Pole"
  231 => "Antarctica/Syowa"
  232 => "Antarctica/Troll"
  233 => "Antarctica/Vostok"
  234 => "Arctic/Longyearbyen"
  235 => "Asia/Aden"
  236 => "Asia/Almaty"
  237 => "Asia/Amman"
  238 => "Asia/Anadyr"
  239 => "Asia/Aqtau"
  240 => "Asia/Aqtobe"
  241 => "Asia/Ashgabat"
  242 => "Asia/Ashkhabad"
  243 => "Asia/Atyrau"
  244 => "Asia/Baghdad"
  245 => "Asia/Bahrain"
  246 => "Asia/Baku"
  247 => "Asia/Bangkok"
  248 => "Asia/Barnaul"
  249 => "Asia/Beirut"
  250 => "Asia/Bishkek"
  251 => "Asia/Brunei"
  252 => "Asia/Calcutta"
  253 => "Asia/Chita"
  254 => "Asia/Choibalsan"
  255 => "Asia/Chongqing"
  256 => "Asia/Chungking"
  257 => "Asia/Colombo"
  258 => "Asia/Dacca"
  259 => "Asia/Damascus"
  260 => "Asia/Dhaka"
  261 => "Asia/Dili"
  262 => "Asia/Dubai"
  263 => "Asia/Dushanbe"
  264 => "Asia/Famagusta"
  265 => "Asia/Gaza"
  266 => "Asia/Harbin"
  267 => "Asia/Hebron"
  268 => "Asia/Ho_Chi_Minh"
  269 => "Asia/Hong_Kong"
  270 => "Asia/Hovd"
  271 => "Asia/Irkutsk"
  272 => "Asia/Istanbul"
  273 => "Asia/Jakarta"
  274 => "Asia/Jayapura"
  275 => "Asia/Jerusalem"
  276 => "Asia/Kabul"
  277 => "Asia/Kamchatka"
  278 => "Asia/Karachi"
  279 => "Asia/Kashgar"
  280 => "Asia/Kathmandu"
  281 => "Asia/Katmandu"
  282 => "Asia/Khandyga"
  283 => "Asia/Kolkata"
  284 => "Asia/Krasnoyarsk"
  285 => "Asia/Kuala_Lumpur"
  286 => "Asia/Kuching"
  287 => "Asia/Kuwait"
  288 => "Asia/Macao"
  289 => "Asia/Macau"
  290 => "Asia/Magadan"
  291 => "Asia/Makassar"
  292 => "Asia/Manila"
  293 => "Asia/Muscat"
  294 => "Asia/Nicosia"
  295 => "Asia/Novokuznetsk"
  296 => "Asia/Novosibirsk"
  297 => "Asia/Omsk"
  298 => "Asia/Oral"
  299 => "Asia/Phnom_Penh"
  300 => "Asia/Pontianak"
  301 => "Asia/Pyongyang"
  302 => "Asia/Qatar"
  303 => "Asia/Qostanay"
  304 => "Asia/Qyzylorda"
  305 => "Asia/Rangoon"
  306 => "Asia/Riyadh"
  307 => "Asia/Saigon"
  308 => "Asia/Sakhalin"
  309 => "Asia/Samarkand"
  310 => "Asia/Seoul"
  311 => "Asia/Shanghai"
  312 => "Asia/Singapore"
  313 => "Asia/Srednekolymsk"
  314 => "Asia/Taipei"
  315 => "Asia/Tashkent"
  316 => "Asia/Tbilisi"
  317 => "Asia/Tehran"
  318 => "Asia/Tel_Aviv"
  319 => "Asia/Thimbu"
  320 => "Asia/Thimphu"
  321 => "Asia/Tokyo"
  322 => "Asia/Tomsk"
  323 => "Asia/Ujung_Pandang"
  324 => "Asia/Ulaanbaatar"
  325 => "Asia/Ulan_Bator"
  326 => "Asia/Urumqi"
  327 => "Asia/Ust-Nera"
  328 => "Asia/Vientiane"
  329 => "Asia/Vladivostok"
  330 => "Asia/Yakutsk"
  331 => "Asia/Yangon"
  332 => "Asia/Yekaterinburg"
  333 => "Asia/Yerevan"
  334 => "Atlantic/Azores"
  335 => "Atlantic/Bermuda"
  336 => "Atlantic/Canary"
  337 => "Atlantic/Cape_Verde"
  338 => "Atlantic/Faeroe"
  339 => "Atlantic/Faroe"
  340 => "Atlantic/Jan_Mayen"
  341 => "Atlantic/Madeira"
  342 => "Atlantic/Reykjavik"
  343 => "Atlantic/South_Georgia"
  344 => "Atlantic/St_Helena"
  345 => "Atlantic/Stanley"
  346 => "Australia/ACT"
  347 => "Australia/Adelaide"
  348 => "Australia/Brisbane"
  349 => "Australia/Broken_Hill"
  350 => "Australia/Canberra"
  351 => "Australia/Currie"
  352 => "Australia/Darwin"
  353 => "Australia/Eucla"
  354 => "Australia/Hobart"
  355 => "Australia/LHI"
  356 => "Australia/Lindeman"
  357 => "Australia/Lord_Howe"
  358 => "Australia/Melbourne"
  359 => "Australia/North"
  360 => "Australia/NSW"
  361 => "Australia/Perth"
  362 => "Australia/Queensland"
  363 => "Australia/South"
  364 => "Australia/Sydney"
  365 => "Australia/Tasmania"
  366 => "Australia/Victoria"
  367 => "Australia/West"
  368 => "Australia/Yancowinna"
  369 => "Brazil/Acre"
  370 => "Brazil/DeNoronha"
  371 => "Brazil/East"
  372 => "Brazil/West"
  373 => "Canada/Atlantic"
  374 => "Canada/Central"
  375 => "Canada/Eastern"
  376 => "Canada/Mountain"
  377 => "Canada/Newfoundland"
  378 => "Canada/Pacific"
  379 => "Canada/Saskatchewan"
  380 => "Canada/Yukon"
  381 => "CET"
  382 => "Chile/Continental"
  383 => "Chile/EasterIsland"
  384 => "CST6CDT"
  385 => "Cuba"
  386 => "EET"
  387 => "Egypt"
  388 => "Eire"
  389 => "EST"
  390 => "EST5EDT"
  391 => "Etc/GMT"
  392 => "Etc/GMT+0"
  393 => "Etc/GMT+1"
  394 => "Etc/GMT+10"
  395 => "Etc/GMT+11"
  396 => "Etc/GMT+12"
  397 => "Etc/GMT+2"
  398 => "Etc/GMT+3"
  399 => "Etc/GMT+4"
  400 => "Etc/GMT+5"
  401 => "Etc/GMT+6"
  402 => "Etc/GMT+7"
  403 => "Etc/GMT+8"
  404 => "Etc/GMT+9"
  405 => "Etc/GMT-0"
  406 => "Etc/GMT-1"
  407 => "Etc/GMT-10"
  408 => "Etc/GMT-11"
  409 => "Etc/GMT-12"
  410 => "Etc/GMT-13"
  411 => "Etc/GMT-14"
  412 => "Etc/GMT-2"
  413 => "Etc/GMT-3"
  414 => "Etc/GMT-4"
  415 => "Etc/GMT-5"
  416 => "Etc/GMT-6"
  417 => "Etc/GMT-7"
  418 => "Etc/GMT-8"
  419 => "Etc/GMT-9"
  420 => "Etc/GMT0"
  421 => "Etc/Greenwich"
  422 => "Etc/UCT"
  423 => "Etc/Universal"
  424 => "Etc/UTC"
  425 => "Etc/Zulu"
  426 => "Europe/Amsterdam"
  427 => "Europe/Andorra"
  428 => "Europe/Astrakhan"
  429 => "Europe/Athens"
  430 => "Europe/Belfast"
  431 => "Europe/Belgrade"
  432 => "Europe/Berlin"
  433 => "Europe/Bratislava"
  434 => "Europe/Brussels"
  435 => "Europe/Bucharest"
  436 => "Europe/Budapest"
  437 => "Europe/Busingen"
  438 => "Europe/Chisinau"
  439 => "Europe/Copenhagen"
  440 => "Europe/Dublin"
  441 => "Europe/Gibraltar"
  442 => "Europe/Guernsey"
  443 => "Europe/Helsinki"
  444 => "Europe/Isle_of_Man"
  445 => "Europe/Istanbul"
  446 => "Europe/Jersey"
  447 => "Europe/Kaliningrad"
  448 => "Europe/Kiev"
  449 => "Europe/Kirov"
  450 => "Europe/Kyiv"
  451 => "Europe/Lisbon"
  452 => "Europe/Ljubljana"
  453 => "Europe/London"
  454 => "Europe/Luxembourg"
  455 => "Europe/Madrid"
  456 => "Europe/Malta"
  457 => "Europe/Mariehamn"
  458 => "Europe/Minsk"
  459 => "Europe/Monaco"
  460 => "Europe/Moscow"
  461 => "Europe/Nicosia"
  462 => "Europe/Oslo"
  463 => "Europe/Paris"
  464 => "Europe/Podgorica"
  465 => "Europe/Prague"
  466 => "Europe/Riga"
  467 => "Europe/Rome"
  468 => "Europe/Samara"
  469 => "Europe/San_Marino"
  470 => "Europe/Sarajevo"
  471 => "Europe/Saratov"
  472 => "Europe/Simferopol"
  473 => "Europe/Skopje"
  474 => "Europe/Sofia"
  475 => "Europe/Stockholm"
  476 => "Europe/Tallinn"
  477 => "Europe/Tirane"
  478 => "Europe/Tiraspol"
  479 => "Europe/Ulyanovsk"
  480 => "Europe/Uzhgorod"
  481 => "Europe/Vaduz"
  482 => "Europe/Vatican"
  483 => "Europe/Vienna"
  484 => "Europe/Vilnius"
  485 => "Europe/Volgograd"
  486 => "Europe/Warsaw"
  487 => "Europe/Zagreb"
  488 => "Europe/Zaporozhye"
  489 => "Europe/Zurich"
  490 => "Factory"
  491 => "GB"
  492 => "GB-Eire"
  493 => "GMT"
  494 => "GMT+0"
  495 => "GMT-0"
  496 => "GMT0"
  497 => "Greenwich"
  498 => "Hongkong"
  499 => "HST"
  500 => "Iceland"
  501 => "Indian/Antananarivo"
  502 => "Indian/Chagos"
  503 => "Indian/Christmas"
  504 => "Indian/Cocos"
  505 => "Indian/Comoro"
  506 => "Indian/Kerguelen"
  507 => "Indian/Mahe"
  508 => "Indian/Maldives"
  509 => "Indian/Mauritius"
  510 => "Indian/Mayotte"
  511 => "Indian/Reunion"
  512 => "Iran"
  513 => "Israel"
  514 => "Jamaica"
  515 => "Japan"
  516 => "Kwajalein"
  517 => "leapseconds"
  518 => "Libya"
  519 => "localtime"
  520 => "MET"
  521 => "Mexico/BajaNorte"
  522 => "Mexico/BajaSur"
  523 => "Mexico/General"
  524 => "MST"
  525 => "MST7MDT"
  526 => "Navajo"
  527 => "NZ"
  528 => "NZ-CHAT"
  529 => "Pacific/Apia"
  530 => "Pacific/Auckland"
  531 => "Pacific/Bougainville"
  532 => "Pacific/Chatham"
  533 => "Pacific/Chuuk"
  534 => "Pacific/Easter"
  535 => "Pacific/Efate"
  536 => "Pacific/Enderbury"
  537 => "Pacific/Fakaofo"
  538 => "Pacific/Fiji"
  539 => "Pacific/Funafuti"
  540 => "Pacific/Galapagos"
  541 => "Pacific/Gambier"
  542 => "Pacific/Guadalcanal"
  543 => "Pacific/Guam"
  544 => "Pacific/Honolulu"
  545 => "Pacific/Johnston"
  546 => "Pacific/Kanton"
  547 => "Pacific/Kiritimati"
  548 => "Pacific/Kosrae"
  549 => "Pacific/Kwajalein"
  550 => "Pacific/Majuro"
  551 => "Pacific/Marquesas"
  552 => "Pacific/Midway"
  553 => "Pacific/Nauru"
  554 => "Pacific/Niue"
  555 => "Pacific/Norfolk"
  556 => "Pacific/Noumea"
  557 => "Pacific/Pago_Pago"
  558 => "Pacific/Palau"
  559 => "Pacific/Pitcairn"
  560 => "Pacific/Pohnpei"
  561 => "Pacific/Ponape"
  562 => "Pacific/Port_Moresby"
  563 => "Pacific/Rarotonga"
  564 => "Pacific/Saipan"
  565 => "Pacific/Samoa"
  566 => "Pacific/Tahiti"
  567 => "Pacific/Tarawa"
  568 => "Pacific/Tongatapu"
  569 => "Pacific/Truk"
  570 => "Pacific/Wake"
  571 => "Pacific/Wallis"
  572 => "Pacific/Yap"
  573 => "Poland"
  574 => "Portugal"
  575 => "PRC"
  576 => "PST8PDT"
  577 => "ROC"
  578 => "ROK"
  579 => "Singapore"
  580 => "Turkey"
  581 => "tzdata.zi"
  582 => "UCT"
  583 => "Universal"
  584 => "US/Alaska"
  585 => "US/Aleutian"
  586 => "US/Arizona"
  587 => "US/Central"
  588 => "US/East-Indiana"
  589 => "US/Eastern"
  590 => "US/Hawaii"
  591 => "US/Indiana-Starke"
  592 => "US/Michigan"
  593 => "US/Mountain"
  594 => "US/Pacific"
  595 => "US/Samoa"
  596 => "UTC"
  597 => "W-SU"
  598 => "WET"
  599 => "Zulu"
]
```

Reply:
`-new` remove it from end